src/PowerShellEditorServices/Services/Symbols/Visitors/SymbolVisitor.cs

```diff
@@ -0,0 +1,231 @@
+    {
+        SymbolType symbolType = functionDefinitionAst.IsWorkflow
+            ? SymbolType.Workflow
+            : SymbolType.Function;
+
+        // Extent for constructors and method trigger both this and VisitFunctionMember(). Covered in the latter.
+        // This will not exclude nested functions as they have ScriptBlockAst as parent
+        if (functionDefinitionAst.Parent is FunctionMemberAst)
+        {
+            return AstVisitAction.Continue;
+        }
+
+        IScriptExtent nameExtent = VisitorUtils.GetNameExtent(functionDefinitionAst);
+        return _action(new SymbolReference(
+            symbolType,
+            functionDefinitionAst.Name,
+            VisitorUtils.GetFunctionDisplayName(functionDefinitionAst),
+            nameExtent,
+            functionDefinitionAst.Extent,
+            _file,
+            isDeclaration: true));
+    }
+
+    public override AstVisitAction VisitParameter(ParameterAst parameterAst)
+    {
+        return _action(new SymbolReference(
+            SymbolType.Parameter,
+            $"${parameterAst.Name.VariablePath.UserPath}",
+            VisitorUtils.GetParamDisplayName(parameterAst),
+            parameterAst.Name.Extent,
+            parameterAst.Extent,
+            _file,
+            isDeclaration: true));
+    }
+
+    public override AstVisitAction VisitVariableExpression(VariableExpressionAst variableExpressionAst)
+    {
+        // TODO: Consider tracking unscoped variable references only when they declared within
+        // the same function definition.
+        return _action(new SymbolReference(
+            SymbolType.Variable,
+            $"${variableExpressionAst.VariablePath.UserPath}",
+            $"${variableExpressionAst.VariablePath.UserPath}",
+            variableExpressionAst.Extent,
+            variableExpressionAst.Extent, // TODO: Maybe parent?
+            _file,
+            isDeclaration: variableExpressionAst.Parent is AssignmentStatementAst or ParameterAst));
+    }
+
+    public override AstVisitAction VisitTypeDefinition(TypeDefinitionAst typeDefinitionAst)
+    {
+        SymbolType symbolType = typeDefinitionAst.IsEnum
+            ? SymbolType.Enum
+            : SymbolType.Class;
+
+        IScriptExtent nameExtent = VisitorUtils.GetNameExtent(typeDefinitionAst);
+        return _action(new SymbolReference(
+            symbolType,
+            typeDefinitionAst.Name,
+            (symbolType is SymbolType.Enum ? "enum " : "class ") + typeDefinitionAst.Name + " { }",
+            nameExtent,
+            typeDefinitionAst.Extent,
+            _file,
+            isDeclaration: true));
+    }
+
+    public override AstVisitAction VisitTypeExpression(TypeExpressionAst typeExpressionAst)
+    {
+        return _action(new SymbolReference(
+            SymbolType.Type,
+            typeExpressionAst.TypeName.Name,
+            typeExpressionAst.TypeName.Name,
```

Comment:
Nit. Same as `VisitTypeConstraint` or change both to `"type" + typeExpressionAst.TypeName.Name`? No preference, but `symbol name<overload>` as pattern in general for displaynames?